docs/content/project/contributing.md

```diff
@@ -0,0 +1,253 @@
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">1</span><a href="#consensus">Discuss</a></h2>
+        <p>Create a Issue or mailing list discussion and reach consensus</p>
+        <p><b>To request an issue, please note that it is not just a "please assign it to me", you need to explain your understanding of the issue, and your design, and if possible, you need to provide your POC code.</b></p>
+      </div>
+    </div>
+  </div>
+  <div class="column">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">2</span><a href="#implement">Implement</a></h2>
+        <p>Create the Pull Request and the approach agreed upon in the issue.</p>
+        <p><b>1.Only create the PR if you are assigned to the issue. 2.Please associate an issue (if any), e.g. fix #123. 3.Please enable the actions of your own clone project, show a screenshot of the test passing.</b></p>
+      </div>
+    </div>
+  </div>
+  <div class="column">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">3</span><a href="#review">Review</a></h2>
+        <p>Work with the reviewer.</p><br />
+        <p><b>1.Make sure no unrelated or unnecessary reformatting changes are included. 2.Make sure your change has been rebased to the latest commits in your base branch.</b></p>
+      </div>
+    </div>
+  </div>
+  <div class="column">
+    <div class="panel panel-default">
+      <div class="panel-body">
+        <h2><span class="number">4</span><a href="#merge">Merge</a></h2>
+        <p>A committer of Paimon checks if the contribution fulfills the requirements and merges the code to the codebase.</p>
+      </div>
+    </div>
+  </div>
+</div>
+
+## Code Review Guide
+
+Every review needs to check the following six aspects. **We encourage to check these aspects in order, to avoid
+spending time on detailed code quality reviews when formal requirements are not met or there is no consensus in
+the community to accept the change.**
+
+#### 1. Is the Contribution Well-Described?
+
+Check whether the contribution is sufficiently well-described to support a good review. Trivial changes and fixes
+do not need a long description. If the implementation is exactly according to a prior discussion on Issue or the
+development mailing list, only a short reference to that discussion is needed.
+
+If the implementation is different from the agreed approach in the consensus discussion, a detailed description of
+the implementation is required for any further review of the contribution.
+
+#### 2. Does the Contribution Need Attention from some Specific Committers?
+
+Some changes require attention and approval from specific committers.
+
+If the pull request needs specific attention, one of the tagged committers/contributors should give the final approval.
+
+#### 3. Is the Overall Code Quality Good, Meeting Standard we Want to Maintain in Paimon?
```

Comment:
I think the description of this part is a bit ambiguous. How can we define good code quality? I think we can add reference to [Flink's code guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-preamble/) before Paimon already supports one.

Reply:
good point, add `Code guidelines can be found in the [Flink Code Style and Quality Guide](https://flink.apache.org/how-to-contribute/code-style-and-quality-preamble/).`